module/jobqueue/README.md

```diff
@@ -0,0 +1,63 @@
+## JobQueue Design Goal
+
+The jobqueue package implemented a reusable job queue system for async message processing.
+
+The most common use case is to work on each finalized block async.
+
+For instance, the verification node verify each finalized block. The block processing needs to be async because if not, the verification node might get overwhelmed by the large amount of blocks to be processed when they got finalized quickly during the period that the node went back online and catching up blocks from other peers.
```

Comment:
>So is the job queue only creating a job for each new finalized block? or can it be used for anything?

The finalized block is the first scenario for the job queue. Since each finalized block can be indexed by height, we use a value uint64 value as job index to find each job. 

So it can be used in other things if jobs can be indexed by uint64 and are consecutive, which means it cannot be used for each incorporated block, because incorporated blocks cannot be indexed by height.